pkg/spanconfig/spanconfigkvsubscriber/datadriven_test.go

```diff
@@ -0,0 +1,317 @@
+//   internals of the kvsubscriber and is useful to test teardown behavior.
+//
+// Text of the form [a,b) and [a,b):C correspond to spans and span config
+// entries; see spanconfigtestutils.Parse{Span,Config,SpanConfigEntry} for more
+// details.
+func TestDataDriven(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	datadriven.Walk(t, testutils.TestDataPath(t), func(t *testing.T, path string) {
+		ctx := context.Background()
+		ctx, cancel := context.WithCancel(ctx)
+		tc := testcluster.StartTestCluster(t, 1, base.TestClusterArgs{
+			ServerArgs: base.TestServerArgs{
+				EnableSpanConfigs: true,
+			},
+		})
+		defer cancel()
+		defer tc.Stopper().Stop(ctx)
+
+		ts := tc.Server(0)
+		tdb := sqlutils.MakeSQLRunner(tc.ServerConn(0))
+		tdb.Exec(t, `SET CLUSTER SETTING kv.rangefeed.enabled = true`)
+		tdb.Exec(t, `SET CLUSTER SETTING spanconfig.experimental_kvaccessor.enabled = true`)
+		tdb.Exec(t, `SET CLUSTER SETTING kv.closed_timestamp.target_duration = '100ms'`)
+
+		const dummyTableName = "dummy_span_configurations"
+		tdb.Exec(t, fmt.Sprintf("CREATE TABLE %s (LIKE system.span_configurations INCLUDING ALL)", dummyTableName))
+
+		var dummyTableID uint32
+		tdb.QueryRow(t, fmt.Sprintf(
+			`SELECT table_id from crdb_internal.tables WHERE name = '%s'`, dummyTableName),
+		).Scan(&dummyTableID)
+
+		kvAccessor := spanconfigkvaccessor.New(
+			tc.Server(0).DB(),
+			tc.Server(0).InternalExecutor().(sqlutil.InternalExecutor),
+			tc.Server(0).ClusterSettings(),
+			fmt.Sprintf("defaultdb.public.%s", dummyTableName),
+		)
+
+		mu := struct {
+			syncutil.Mutex
+			lastFrontierTS    hlc.Timestamp // serializes kvsubscriber-updates and kvaccessor-update
+			subscriberRunning bool          // serializes kvsubscriber-{updates,subscribe,inject-buffer-overflow}
+			receivedUpdates   roachpb.Spans
+		}{}
+		injectedErrCh := make(chan error)
+
+		kvSubscriber := spanconfigkvsubscriber.New(
+			ts.Stopper(),
+			ts.DB(),
+			ts.Clock(),
+			ts.RangeFeedFactory().(*rangefeed.Factory),
+			dummyTableID,
+			10<<20, /* 10 MB */
+			spanconfigtestutils.ParseConfig(t, "MISSING"),
+			&spanconfig.TestingKnobs{
+				KVSubscriberOnTimestampAdvanceInterceptor: func(ts hlc.Timestamp) {
+					mu.Lock()
+					defer mu.Unlock()
+					mu.lastFrontierTS = ts
+				},
+				KVSubscriberPostRangefeedStartInterceptor: func() {
+					mu.Lock()
+					defer mu.Unlock()
+					mu.subscriberRunning = true
+				},
+				KVSubscriberPreExitInterceptor: func() {
+					mu.Lock()
+					defer mu.Unlock()
+					mu.subscriberRunning = false
+				},
+				KVSubscriberErrorInjectionCh: injectedErrCh,
+			},
+		)
+
+		kvSubscriber.OnSpanConfigUpdate(ctx, func(span roachpb.Span) {
+			mu.Lock()
+			defer mu.Unlock()
+			mu.receivedUpdates = append(mu.receivedUpdates, span)
+		})
+
+		var lastUpdateTS hlc.Timestamp
+		datadriven.RunTest(t, path, func(t *testing.T, d *datadriven.TestData) string {
+			switch d.Cmd {
+			case "kvaccessor-get":
```

Comment:
nit: Do you find the `kvaccessor` and `kvsubscriber` prefixes are valuable? I don't see any attempt to compose the various data driven tests. It feels to me like `get`/`update`/`subscribe`/`updates` etc would be clear enough.